DB/src/main/java/io/deephaven/db/tables/utils/CsvHelpers.java

```diff
@@ -575,6 +598,110 @@ public static QueryTable readCsv2(InputStream is) throws IOException {
+        }
+        if (fileName.endsWith(".bz2")) {
+            return new BZip2CompressorInputStream(Files.newInputStream(path));
+        }
+        if (fileName.endsWith(".gz")) {
+            return new GZIPInputStream(Files.newInputStream(path));
+        }
+        if (fileName.endsWith(".7z")) {
+            final SevenZipInputStream stream = new SevenZipInputStream(new InputStreamFactory() {
+                @Override
+                public InputStream createInputStream() throws IOException {
+                    return Files.newInputStream(path);
+                }
+
+                @Override
+                public String getDescription() {
+                    return path.toString();
+                }
+            });
+            stream.getNextEntry(Behavior.SKIP_WHEN_NO_STREAM);
+            return stream;
+        }
+        if (fileName.endsWith(".zst")) {
+            return new ZstdCompressorInputStream(Files.newInputStream(path));
+        }
+        return Files.newInputStream(path);
+    }
+
+    /**
+     * Creates an in-memory table from {@code file} according to the {@code specs}.
+     *
+     * <p>
+     * Paths that end in ".zip", ".bz2", ".gz", ".7z", or ".zst" will be decompressed.
+     *
+     * @param file the file
+     * @param specs the csv specs
+     * @return the table
+     * @throws IOException if an I/O exception occurs
+     * @see #open(Path)
+     */
+    @ScriptApi
+    public static Table readCsv(String file, CsvSpecs specs) throws IOException {
+        return readCsv(Paths.get(file), specs);
+    }
+
+    /**
+     * Creates an in-memory table from {@code path} according to the {@code specs}.
+     *
+     * <p>
+     * Paths that end in ".zip", ".bz2", ".gz", ".7z", or ".zst" will be decompressed.
+     *
+     * @param path the path
+     * @param specs the csv specs
+     * @return the table
+     * @throws IOException if an I/O exception occurs
+     * @see #open(Path)
+     */
+    @ScriptApi
+    public static Table readCsv(Path path, CsvSpecs specs) throws IOException {
+        return InMemoryTable.from(specs.parse(open(path)));
+    }
+
+    /**
+     * Creates an in-memory table from {@code url} according to the {@code specs}.
+     *
+     * @param url the url
+     * @param specs the csv specs
+     * @return the table
+     * @throws IOException if an I/O exception occurs
+     */
+    @ScriptApi
+    public static Table readCsv(URL url, CsvSpecs specs) throws IOException {
+        return InMemoryTable.from(specs.parse(url.openStream()));
+    }
+
+    /**
+     * Creates an in-memory table from {@code stream} according to the {@code specs}. The {@code stream} will be closed
+     * upon return.
+     *
+     * @param stream the stream
+     * @param specs the csv specs
+     * @return the table
+     * @throws IOException if an I/O exception occurs
+     */
+    @ScriptApi
+    public static Table readCsv(InputStream stream, CsvSpecs specs) throws IOException {
```

Comment:
We should either deprecate, delete, or re-implement the old ones in terms of the new ones.